torch/csrc/autograd/python_variable.cpp

```diff
@@ -459,6 +473,42 @@ static PyObject * THPVariable_device(THPVariable* self) {
   END_HANDLE_TH_ERRORS
 }
 
+// Returns the type object of the most derived subclass of torch._C._TensorBase that is not an instance of torch.nn.Parameter.
+PyTypeObject *THPVariable_result_ptype(PyObject *self, PyObject *args) {
+    HANDLE_TH_ERRORS
+    PyTypeObject *ptype;
+    PyObject *obj;
+
+    //Initialize on First Call
+    bool initialized = false;
+    static PyObject *base, *parameter;
+    if (!initialized){
+        base = THPVariableClass;
+        parameter = PyObject_GetAttrString(PyImport_ImportModule("torch.nn.parameter"), "Parameter");
+        if (parameter == nullptr)
+            return nullptr;
+        initialized = true;
+    }
+
+    ptype = (PyTypeObject*)base;
+    Py_INCREF(ptype);
+    // torch.nn.Parameter explicitly do not perform ptype propagation so this should not effect torch.nn modules.
+    if (self != nullptr && PyObject_IsInstance(self, base) && !PyObject_IsInstance(self, parameter) && PyObject_IsInstance(self, (PyObject*)ptype)){
```

Comment:
Can we prefer explicit comparisons on `o->ob_type` instead of `IsInstance` calls? Those are very slow, and this executed in the hot path.